# 评估和清理热门电子游戏数据

## 分析目标

此数据分析的目的是根据1980年至2023年热门电子游戏数据，通过对该种类游戏用户数量、评论数量等多个维度挖掘玩家普遍喜爱游戏类型。

本项目的目的在于练习评估数据干净和整洁度，并且基于评估结果，对数据进行清洗，从而得到可供下一步分析的数据。

## 简介

数据集包含从 1980 年到 2023 年的视频游戏列表，并提供发布日期、用户评论评级和评论家评论评级等信息。游戏发布的平台不唯一，且游戏平台使用玩家数量亦有差距，数据选取均通过多种评价综合总结得出的热门电子游戏。

每列数据的含义如下：
- `Title`：游戏标题
- `Release Date`：游戏首个版本发布日期
- `Team`：游戏开发团队
- `Rating`：平均评分
- `Times Listed`：列出此游戏的用户数量
- `Number of Reviews`：用户提供的评论数量
- `Genres`：游戏所属的所有类型/流派
- `Summary`：团队提供的摘要/概述
- `Reviews`：用户的评价/评论

## 读取数据

导入数据分析所需要的库，并通过Pandas的`read_csv`函数，将原始数据文件"games.csv"里的数据内容，解析为DataFrame，并赋值给变量`original_data`。

In [1]:
import pandas as pd

In [2]:
original_data = pd.read_csv("games.csv")

In [3]:
original_data.head()

Unnamed: 0                                    Title  Release Date  \
0           0                               Elden Ring  Feb 25, 2022   
1           1                                    Hades  Dec 10, 2019   
2           2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3           3                                Undertale  Sep 15, 2015   
4           4                            Hollow Knight  Feb 24, 2017   

                                                Team  Rating Times Listed  \
0     ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                               ['Supergiant Games']     4.3         2.9K   
2  ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                 ['tobyfox', '8-4']     4.2         3.5K   
4                                    ['Team Cherry']     4.4           3K   

  Number of Reviews                                             Genres  \
0              3.9K                               ['Adventure', 'RPG']   
1              2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2              4.3K                               ['Adventure', 'RPG']   
3              3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                3K                 ['Adventure', 'Indie', 'Platform']   

                                             Summary  \
0  Elden Ring is a fantasy, action and open world...   
1  A rogue-lite hack and slash dungeon crawler in...   
2  The Legend of Zelda: Breath of the Wild is the...   
3  A small child falls into the Underground, wher...   
4  A 2D metroidvania with an emphasis on close co...   

                                             Reviews Plays Playing Backlogs  \
0  ["The first playthrough of elden ring is one o...   17K    3.8K     4.6K   
1  ['convinced this is a roguelike for people who...   21K    3.2K     6.3K   
2  ['This game is the game (that is not CS:GO) th...   30K    2.5K       5K   
3  ['soundtrack is tied for #1 with nier automata...   28K     679     4.9K   
4  ["this games worldbuilding is incredible, with...   21K    2.4K     8.3K   

  Wishlist  
0     4.8K  
1     3.6K  
2     2.6K  
3     1.8K  
4     2.3K

## 评估数据

在这一部分，我将对在上一部分建立的`original_data`这个DataFrame所包含的数据进行评估。

评估主要从两个方面进行：结构和内容，即整齐度和干净度。数据的结构性问题指不符合“每列是一个变量，每行是一个观察值，每个单元格是一个值”这三个标准，数据的内容性问题包括存在丢失数据、重复数据、无效数据等。

### 评估数据整齐度

In [4]:
original_data.sample(10)

Unnamed: 0                                Title  Release Date  \
1431        1431       Silent Hill 2: Restless Dreams  Dec 19, 2001   
667          667                         Wanted: Dead  Feb 14, 2023   
531          531                    Metal: Hellsinger  Sep 15, 2022   
1084        1084                Kirby: Planet Robobot  Apr 28, 2016   
953          953           Uncharted 2: Among Thieves  Oct 13, 2009   
675          675                        Sonic Advance  Dec 20, 2001   
786          786                      Persona 5 Royal  Oct 31, 2019   
743          743  Fallout 3: Game of the Year Edition  Oct 13, 2009   
131          131                          Titanfall 2  Oct 28, 2016   
1499        1499           WWE SmackDown vs. Raw 2008  Nov 09, 2007   

                                                 Team  Rating Times Listed  \
1431                        ['Team Silent', 'Konami']     4.5           90   
667                      ['Soleil', '110 Industries']     2.6           81   
531               ['Level Infinite', 'The Outsiders']     3.7          322   
1084                   ['HAL Laboratory', 'Nintendo']     4.2          669   
953    ['Naughty Dog', 'Sony Computer Entertainment']     3.9         1.4K   
675                                 ['Sega', 'Dimps']     3.2          446   
786                            ['Atlus USA', 'Atlus']     4.4         2.7K   
743   ['Bethesda Game Studios', 'Bethesda Softworks']     3.6          188   
131      ['Respawn Entertainment', 'Electronic Arts']     4.1         1.3K   
1499                      ['THQ', "YUKE'S Co., Ltd."]     3.1           62   

     Number of Reviews                                       Genres  \
1431                90                                ['Adventure']   
667                 81          ['Adventure', 'Brawler', 'Shooter']   
531                322            ['Adventure', 'Music', 'Shooter']   
1084               669                    ['Adventure', 'Platform']   
953               1.4K         ['Adventure', 'Platform', 'Shooter']   
675                446                                 ['Platform']   
786               2.7K  ['Adventure', 'RPG', 'Turn Based Strategy']   
743                188              ['Adventure', 'RPG', 'Shooter']   
131               1.3K                     ['Adventure', 'Shooter']   
1499                62                        ['Fighting', 'Sport']   

                                                Summary  \
1431  Silent Hill 2: Restless Dreams is an updated v...   
667   Wanted: Dead promises "spectacular melee comba...   
531   Strike terror into the hearts of demons and de...   
1084  "Harness the power of a mysterious mech to sto...   
953   In the sequel to Drake's Fortune, Nathan Drake...   
675   Sonic Advance is notable for being the first S...   
786   An enhanced version of Persona 5 with some new...   
743   Prepare for the Future with Fallout 3: Game of...   
131   Titanfall 2 will deliver a crafted experience ...   
1499  The 2008 edition in the Smackdown vs. Raw seri...   

                                                Reviews Plays Playing  \
1431  ['*born from a wish', 'pog', 'Best psychologic...   514      14   
667   ["It's really weird that the narrative around ...   100      15   
531   ['Should have won best OST 2022', "May revisit...  1.2K      97   
1084  ['Remaster plz', 'Kirby can (yet again) do no ...  3.6K      82   
953   ["Infinitely better than the first game! The p...   12K      96   
675   ['Disappointing in every way. Mediocre music, ...    4K      26   
786   ['Verdadeiro goty 2017, zelda é o caralho. Vai...   12K    2.3K   
743   ['O primeiro jogo da franquia do Fallout a ser...  2.6K      67   
131   ['iyi', "It's a short game and all but I had f...   12K     163   
1499  ['The earliest memory I have of this game was ...  1.4K       1   

     Backlogs Wishlist  
1431      130       81  
667        94      212  
531       419      424  
1084       1K      644  
953 

从抽样的10行数据数据来看，数据符合“每列是一个变量，每行是一个观察值，每个单元格是一个值”，具体来看每行是关于某个游戏的名称、简介等方面的介绍，每列则是游戏相关的数据变量，因此不存在结构性问题。

### 评估数据干净度

In [5]:
original_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1512 entries, 0 to 1511
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1512 non-null   int64  
 1   Title              1512 non-null   object 
 2   Release Date       1512 non-null   object 
 3   Team               1511 non-null   object 
 4   Rating             1499 non-null   float64
 5   Times Listed       1512 non-null   object 
 6   Number of Reviews  1512 non-null   object 
 7   Genres             1512 non-null   object 
 8   Summary            1511 non-null   object 
 9   Reviews            1512 non-null   object 
 10  Plays              1512 non-null   object 
 11  Playing            1512 non-null   object 
 12  Backlogs           1512 non-null   object 
 13  Wishlist           1512 non-null   object 
dtypes: float64(1), int64(1), object(12)
memory usage: 165.5+ KB


从输出结果来看，数据共有1512条观察值，而`Team`、`Rating`、`Summary`变量存在缺失值。
`Unnamed:0`列表名不规范，应当进行列名更正。

#### 评估缺失数据

在了解`Team`变量存在缺失值，根据条件提取出缺失观察值并进行分析。

In [6]:
original_data[original_data["Team"].isnull()]

Unnamed: 0                 Title  Release Date Team  Rating  \
1245        1245  NEET Girl Date Night  Oct 21, 2022  NaN     2.7   

     Times Listed Number of Reviews            Genres  \
1245           21                21  ['Visual Novel']   

                                                Summary  \
1245  Your friend sets you up on a date with his NEE...   

                                                Reviews Plays Playing  \
1245  ['this sucked. "Omg she is literally me" is no...   106       1   

     Backlogs Wishlist  
1245       44       42

只有一条游戏数据缺失`Team`变量值。
由于我们分析目的是挖掘玩家喜爱游戏类型，该变量值缺失不会影响到我们的分析目的，且游戏数据其他变量值完整，应予以保留。

此外，`Summary`变量存在缺失值且只有一条，提取该缺失观察值。
`Summary`变量值缺失不会影响到我们的分析目标，但`Rating`变量值同样存在缺失，我们之后进行评估。

In [7]:
original_data[original_data["Summary"].isnull()]

Unnamed: 0              Title     Release Date                    Team  \
649         649  Death Stranding 2  releases on TBD  ['Kojima Productions']   

     Rating Times Listed Number of Reviews                    Genres Summary  \
649     NaN          105               105  ['Adventure', 'Shooter']     NaN   

    Reviews Plays Playing Backlogs Wishlist  
649      []     3       0      209      644

同样`Rating`变量也存在缺失值，采用相同的办法提取出观察值。

In [8]:
original_data[original_data["Rating"].isnull()]

Unnamed: 0                                              Title  \
587          587                                  Final Fantasy XVI   
649          649                                  Death Stranding 2   
713          713                          Final Fantasy VII Rebirth   
719          719                                          Lies of P   
726          726                                              Judas   
746          746  Like a Dragon Gaiden: The Man Who Erased His Name   
972          972          The Legend of Zelda: Tears of the Kingdom   
1130        1130                           Star Wars Jedi: Survivor   
1160        1160            We Love Katamari Reroll + Royal Reverie   
1202        1202                                         Earthblade   
1305        1305                                          Diablo IV   
1341        1341                                   Pac-Man in a Tin   
1349        1349                                       Pokémon Añil   

         Release Date                                               Team  \
587      Jun 22, 2023  ['Square Enix', 'Square Enix Creative Business...   
649   releases on TBD                             ['Kojima Productions']   
713      Dec 31, 2023                                    ['Square Enix']   
719      Aug 01, 2023                        ['NEOWIZ', 'Round8 Studio']   
726      Mar 31, 2025                              ['Ghost Story Games']   
746      Dec 31, 2023                  ['Ryū Ga Gotoku Studios', 'Sega']   
972      May 12, 2023  ['Nintendo', 'Nintendo EPD Production Group No...   
1130     Apr 28, 2023       ['Respawn Entertainment', 'Electronic Arts']   
1160     Jun 02, 2023  ['Bandai Namco Entertainment', 'MONKEYCRAFT Co...   
1202     Dec 31, 2024                             ['Extremely OK Games']   
1305     Jun 06, 2023                         ['Blizzard Entertainment']   
1341     Mar 22, 2023                                 ['Fizz Creations']   
1349     Mar 23, 2023                                      ['Eric Lost']   

      Rating Times Listed Number of Reviews                           Genres  \
587      NaN          422               422                          ['RPG']   
649      NaN          105               105         ['Adventure', 'Shooter']   
713      NaN          192               192                               []   
719      NaN          175               175                          ['RPG']   
726      NaN           90                90         ['Adventure', 'Shooter']   
746      NaN          118               118  ['Adventure', 'Brawler', 'RPG']   
972      NaN          581               581             ['Adventure', 'RPG']   
1130     NaN          250               250                    ['Adventure']   
1160     NaN           51                51          ['Adventure', 'Puzzle']   
1202     NaN           83                83    ['Adventure', 'Indie', 'RPG']   
1305     NaN          148               148               ['Brawler', 'RPG']   
1341     NaN            0                 0                       ['Arcade']   
1349     NaN            2                 2                          ['RPG']   

                                                Summary  \
587   Final Fantasy XVI is an upcoming action role-p...   
649                                                 NaN   
713   This next standalone chapter in the FINAL FANT...   
719   Inspired by the familiar story of Pinocchio, L...   
726   A disintegrating starship. A desperate escape ...   
746   This game covers Kiryu's story between Yakuza ...   
972   The Legend of Zelda: Tears of the Kingdom is t...   
1130  The story of Cal Kestis continues in Star Wars...   
1160  We Love Katamari Reroll + Royal Reverie is a r...   
1202  You are Névoa, an enigmatic child of Fate retu...   
1305  Endless demons to slaughter. Deep customizatio...   
1341  Pac-Man in a Tin is a pocket-sized tin contain...   
1349  Pokémon Añil is the fourth video game created ...   

有13条游戏数据缺失`Rating`变量值。

从输出的结果来看，这些缺失`Rating`的游戏数据的发行日期并不一定是2023年年末发行而导致变量值缺失，存在其他情况导致数据异常，而游戏评分是我们分析玩家热爱游戏类型的重要指标之一，认为这些数据无法提供有效含义，因此后续可以进行删除。

#### 评估重复数据

根据数据变量含义来看，`Title`,`Team`都是唯一标识符，但是一个团队可以发行多种游戏，`Team`可以存在重复，而游戏名也能存在重复，但是两者不能同时重复。

所以针对该数据集，我们需要检查是否存在变量值`Title`,`Team`都存在重复的数据。

In [10]:
original_data[original_data.duplicated(subset = ['Title','Team'])]

Unnamed: 0                                    Title  Release Date  \
326          326                               Elden Ring  Feb 25, 2022   
327          327                                    Hades  Dec 10, 2019   
328          328  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
329          329                                Undertale  Sep 15, 2015   
330          330                            Hollow Knight  Feb 24, 2017   
...          ...                                      ...           ...   
1269        1269                     Final Fantasy XIII-2  Dec 15, 2011   
1270        1270                                  Agar.io  Apr 28, 2015   
1271        1271        Fatal Frame II: Crimson Butterfly  Nov 27, 2003   
1282        1282                     Super Mario Sunshine  Sep 18, 2020   
1496        1496                                  FIFA 13  Sep 25, 2012   

                                                   Team  Rating Times Listed  \
326      ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
327                                ['Supergiant Games']     4.3         2.9K   
328   ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
329                                  ['tobyfox', '8-4']     4.2         3.5K   
330                                     ['Team Cherry']     4.4           3K   
...                                                 ...     ...          ...   
1269                                    ['Square Enix']     3.3          482   
1270              ['Miniclip.com', 'Matheus Valadares']     2.2           81   
1271       ['Tecmo Co., Ltd.', 'Ubisoft Entertainment']     4.2          398   
1282                       ['Nintendo EAD', 'Nintendo']     3.7           19   
1496                   ['Electronic Arts', 'EA Canada']     3.2           62   

     Number of Reviews                                             Genres  \
326               3.9K                               ['Adventure', 'RPG']   
327               2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
328               4.3K                               ['Adventure', 'RPG']   
329               3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
330                 3K                 ['Adventure', 'Indie', 'Platform']   
...                ...                                                ...   
1269               482                               ['Adventure', 'RPG']   
1270                81                              ['Indie', 'Strategy']   
1271               398                                      ['Adventure']   
1282                19                          ['Adventure', 'Platform']   
1496                62                             ['Simulator', 'Sport']   

                                                Summary  \
326   Elden Ring is a fantasy, action and open world...   
327   A rogue-lite hack and slash dungeon crawler in...   
328   The Legend of Zelda: Breath of the Wild is the...   
329   A small child falls into the Underground, wher...   
330   A 2D metroidvania with an emphasis on close co...   
...                                                 ...   
1269  FINAL FANTASY XIII-2 is created with the aim o...   
1270  Agar.io is a Massively-multiplayer top-down st...   
1271  Crimson Butterfly is the second installment in...   
1282  A port of Super Mario Sunshine included in Sup...   
1496  FIFA 13 captures the drama and unpredictabilit...   

                                                Reviews Plays Playing  \
326   ["The first playthrough of elden ring is one o...   17K    3.8K   
327   ['convinced this is a roguelike for people who...   21K    3.2K   
328   ['This game is the game (that is not CS:GO) th...   30K    2.5K   
329   ['soundtrack is tied for #1 with nier automata...   28K     679   
330   ["this games worldbuilding is incredible, with...   21K    2.4K   
...                                                 ...   ...     ...   

从输出结果来看，有398条`Title`,`Team`变量值同时相同的游戏数据，虽然分析目的不涉及到对该游戏类型下游戏评分、游玩人数、评分数量的数值计算，但重复数据会导致数据集冗余，因此在之后需要进行删除。

#### 评估不一致数据

针对此数据集，不存在多个变量指代一个游戏发行公司现象，因此无需进行评估。

#### 评估无效或错误数据

可以通过DataFrame的`describe`方法，对数值统计信息进行快速了解。

In [16]:
original_data.describe()

Unnamed: 0       Rating
count  1512.000000  1499.000000
mean    755.500000     3.719346
std     436.621117     0.532608
min       0.000000     0.700000
25%     377.750000     3.400000
50%     755.500000     3.800000
75%    1133.250000     4.100000
max    1511.000000     4.800000

从输出结果看，`Rating`数据暂未出现异常或负数，因此大致可以认定该数据集不存在无效或错误数据。

此外，`Reviews`对于我们分析目的并无大用且为了保持数据集的直观、简洁，可进行删除操作。

## 清理数据

根据前面评估部分得到的结论，我们需要进行数据的清理包括：
- 把列名`Unnamed: 0`进行纠正
- 把`Rating`缺失的观察值删除
- 把同时存在`Title`,`Team`相同的观察值删除
- 把列`Reviews`删除

为了区分开经过清理的数据和原始的数据，我们创建新的变量`cleaned_data`，让它为`original_data`复制出的副本。我们之后的清理步骤都将被运用在`cleaned_data`上。

In [21]:
cleaned_data = original_data.copy()
cleaned_data.head()

Unnamed: 0                                    Title  Release Date  \
0           0                               Elden Ring  Feb 25, 2022   
1           1                                    Hades  Dec 10, 2019   
2           2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3           3                                Undertale  Sep 15, 2015   
4           4                            Hollow Knight  Feb 24, 2017   

                                                Team  Rating Times Listed  \
0     ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                               ['Supergiant Games']     4.3         2.9K   
2  ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                 ['tobyfox', '8-4']     4.2         3.5K   
4                                    ['Team Cherry']     4.4           3K   

  Number of Reviews                                             Genres  \
0              3.9K                               ['Adventure', 'RPG']   
1              2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2              4.3K                               ['Adventure', 'RPG']   
3              3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                3K                 ['Adventure', 'Indie', 'Platform']   

                                             Summary  \
0  Elden Ring is a fantasy, action and open world...   
1  A rogue-lite hack and slash dungeon crawler in...   
2  The Legend of Zelda: Breath of the Wild is the...   
3  A small child falls into the Underground, wher...   
4  A 2D metroidvania with an emphasis on close co...   

                                             Reviews Plays Playing Backlogs  \
0  ["The first playthrough of elden ring is one o...   17K    3.8K     4.6K   
1  ['convinced this is a roguelike for people who...   21K    3.2K     6.3K   
2  ['This game is the game (that is not CS:GO) th...   30K    2.5K       5K   
3  ['soundtrack is tied for #1 with nier automata...   28K     679     4.9K   
4  ["this games worldbuilding is incredible, with...   21K    2.4K     8.3K   

  Wishlist  
0     4.8K  
1     3.6K  
2     2.6K  
3     1.8K  
4     2.3K

把列名`Unnamed: 0`进行纠正.

In [23]:
cleaned_data = cleaned_data.rename(columns={'Unnamed: 0':'Unnamed'})
cleaned_data

Unnamed                                    Title  Release Date  \
0           0                               Elden Ring  Feb 25, 2022   
1           1                                    Hades  Dec 10, 2019   
2           2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3           3                                Undertale  Sep 15, 2015   
4           4                            Hollow Knight  Feb 24, 2017   
...       ...                                      ...           ...   
1507     1507             Back to the Future: The Game  Dec 22, 2010   
1508     1508                        Team Sonic Racing  May 21, 2019   
1509     1509                           Dragon's Dogma  May 22, 2012   
1510     1510                          Baldur's Gate 3  Oct 06, 2020   
1511     1511                 The LEGO Movie Videogame  Feb 04, 2014   

                                                   Team  Rating Times Listed  \
0        ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                                  ['Supergiant Games']     4.3         2.9K   
2     ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                    ['tobyfox', '8-4']     4.2         3.5K   
4                                       ['Team Cherry']     4.4           3K   
...                                                 ...     ...          ...   
1507                                 ['Telltale Games']     3.2           94   
1508                           ['Sumo Digital', 'Sega']     2.9          264   
1509                                         ['Capcom']     3.7          210   
1510                                 ['Larian Studios']     4.1          165   
1511                          ['WB Games', 'TT Fusion']     2.8          184   

     Number of Reviews                                             Genres  \
0                 3.9K                               ['Adventure', 'RPG']   
1                 2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2                 4.3K                               ['Adventure', 'RPG']   
3                 3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                   3K                 ['Adventure', 'Indie', 'Platform']   
...                ...                                                ...   
1507                94                   ['Adventure', 'Point-and-Click']   
1508               264                               ['Arcade', 'Racing']   
1509               210                                 ['Brawler', 'RPG']   
1510               165  ['Adventure', 'RPG', 'Strategy', 'Tactical', '...   
1511               184                            ['Adventure', 'Puzzle']   

                                                Summary  \
0     Elden Ring is a fantasy, action and open world...   
1     A rogue-lite hack and slash dungeon crawler in...   
2     The Legend of Zelda: Breath of the Wild is the...   
3     A small child falls into the Underground, wher...   
4     A 2D metroidvania with an emphasis on close co...   
...                                                 ...   
1507  Back to the Future: The Game is one of Telltal...   
1508  Team Sonic Racing combines the best elements o...   
1509  Set in a huge open world, Dragon’s Dogma: Dark...   
1510  An ancient evil has returned to Baldur's Gate,...   
1511  Join Emmet and an unlikely group of resistance...   

                                                Reviews Plays Playing  \
0     ["The first playthrough of elden ring is one o...   17K    3.8K   
1     ['convinced this is a roguelike for people who...   21K    3.2K   
2     ['This game is the game (that is not CS:GO) th...   30K    2.5K   
3     ['soundtrack is tied for #1 with nier automata...   28K     679   
4     ["this games worldbuilding is incredible, with...   21K    2.4K   
...                                                 ...   ...     ...   
1507  ['Very enjoyable game. The st

把`Rating`缺失的观察值删除,并查看删除后该列空缺值个数和：

In [24]:
cleaned_data.dropna(subset = ['Rating'],inplace = True)

In [26]:
cleaned_data['Rating'].isnull().sum()

0

把同时存在`Title`,`Team`相同的观察值删除,并检查是否还存在重复数据：

In [27]:
cleaned_data = cleaned_data.drop_duplicates(subset = ['Title','Team'],keep='first')

In [29]:
cleaned_data[cleaned_data.duplicated(subset = ['Title','Team'])]

Empty DataFrame
Columns: [Unnamed, Title, Release Date, Team, Rating, Times Listed, Number of Reviews, Genres, Summary, Reviews, Plays, Playing, Backlogs, Wishlist]
Index: []

把列`Reviews`删除。

In [31]:
cleaned_data = cleaned_data.drop('Reviews',axis = 1)
cleaned_data

Unnamed                                    Title  Release Date  \
0           0                               Elden Ring  Feb 25, 2022   
1           1                                    Hades  Dec 10, 2019   
2           2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3           3                                Undertale  Sep 15, 2015   
4           4                            Hollow Knight  Feb 24, 2017   
...       ...                                      ...           ...   
1507     1507             Back to the Future: The Game  Dec 22, 2010   
1508     1508                        Team Sonic Racing  May 21, 2019   
1509     1509                           Dragon's Dogma  May 22, 2012   
1510     1510                          Baldur's Gate 3  Oct 06, 2020   
1511     1511                 The LEGO Movie Videogame  Feb 04, 2014   

                                                   Team  Rating Times Listed  \
0        ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                                  ['Supergiant Games']     4.3         2.9K   
2     ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                    ['tobyfox', '8-4']     4.2         3.5K   
4                                       ['Team Cherry']     4.4           3K   
...                                                 ...     ...          ...   
1507                                 ['Telltale Games']     3.2           94   
1508                           ['Sumo Digital', 'Sega']     2.9          264   
1509                                         ['Capcom']     3.7          210   
1510                                 ['Larian Studios']     4.1          165   
1511                          ['WB Games', 'TT Fusion']     2.8          184   

     Number of Reviews                                             Genres  \
0                 3.9K                               ['Adventure', 'RPG']   
1                 2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2                 4.3K                               ['Adventure', 'RPG']   
3                 3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                   3K                 ['Adventure', 'Indie', 'Platform']   
...                ...                                                ...   
1507                94                   ['Adventure', 'Point-and-Click']   
1508               264                               ['Arcade', 'Racing']   
1509               210                                 ['Brawler', 'RPG']   
1510               165  ['Adventure', 'RPG', 'Strategy', 'Tactical', '...   
1511               184                            ['Adventure', 'Puzzle']   

                                                Summary Plays Playing  \
0     Elden Ring is a fantasy, action and open world...   17K    3.8K   
1     A rogue-lite hack and slash dungeon crawler in...   21K    3.2K   
2     The Legend of Zelda: Breath of the Wild is the...   30K    2.5K   
3     A small child falls into the Underground, wher...   28K     679   
4     A 2D metroidvania with an emphasis on close co...   21K    2.4K   
...                                                 ...   ...     ...   
1507  Back to the Future: The Game is one of Telltal...   763       5   
1508  Team Sonic Racing combines the best elements o...  1.5K      49   
1509  Set in a huge open world, Dragon’s Dogma: Dark...  1.1K      45   
1510  An ancient evil has returned to Baldur's Gate,...   269      79   
1511  Join Emmet and an unlikely group of resistance...  1.7K      11   

     Backlogs Wishlist  
0        4.6K     4.8K  
1        6.3K     3.6K  
2          5K     2.6K  
3        4.9K     1.8K  
4        8.3K     2.3K  
...       ...      ...  
1507      223       67  
1508      413      107  
1509      487      206  
1510      388      602  
1511      239       73  

[1101 rows x 13 columns]

## 保存清理后的数据

完成数据清理后，把干净整齐的数据保存到新的文件里，文件名为`games_cleaned.csv`。

In [33]:
cleaned_data.head()

Unnamed                                    Title  Release Date  \
0        0                               Elden Ring  Feb 25, 2022   
1        1                                    Hades  Dec 10, 2019   
2        2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3        3                                Undertale  Sep 15, 2015   
4        4                            Hollow Knight  Feb 24, 2017   

                                                Team  Rating Times Listed  \
0     ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                               ['Supergiant Games']     4.3         2.9K   
2  ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                 ['tobyfox', '8-4']     4.2         3.5K   
4                                    ['Team Cherry']     4.4           3K   

  Number of Reviews                                             Genres  \
0              3.9K                               ['Adventure', 'RPG']   
1              2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2              4.3K                               ['Adventure', 'RPG']   
3              3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                3K                 ['Adventure', 'Indie', 'Platform']   

                                             Summary Plays Playing Backlogs  \
0  Elden Ring is a fantasy, action and open world...   17K    3.8K     4.6K   
1  A rogue-lite hack and slash dungeon crawler in...   21K    3.2K     6.3K   
2  The Legend of Zelda: Breath of the Wild is the...   30K    2.5K       5K   
3  A small child falls into the Underground, wher...   28K     679     4.9K   
4  A 2D metroidvania with an emphasis on close co...   21K    2.4K     8.3K   

  Wishlist  
0     4.8K  
1     3.6K  
2     2.6K  
3     1.8K  
4     2.3K

In [34]:
cleaned_data.to_csv("games_cleaned.csv", index=False)

In [35]:
pd.read_csv("games_cleaned.csv").head()

Unnamed                                    Title  Release Date  \
0        0                               Elden Ring  Feb 25, 2022   
1        1                                    Hades  Dec 10, 2019   
2        2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3        3                                Undertale  Sep 15, 2015   
4        4                            Hollow Knight  Feb 24, 2017   

                                                Team  Rating Times Listed  \
0     ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                               ['Supergiant Games']     4.3         2.9K   
2  ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                 ['tobyfox', '8-4']     4.2         3.5K   
4                                    ['Team Cherry']     4.4           3K   

  Number of Reviews                                             Genres  \
0              3.9K                               ['Adventure', 'RPG']   
1              2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2              4.3K                               ['Adventure', 'RPG']   
3              3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                3K                 ['Adventure', 'Indie', 'Platform']   

                                             Summary Plays Playing Backlogs  \
0  Elden Ring is a fantasy, action and open world...   17K    3.8K     4.6K   
1  A rogue-lite hack and slash dungeon crawler in...   21K    3.2K     6.3K   
2  The Legend of Zelda: Breath of the Wild is the...   30K    2.5K       5K   
3  A small child falls into the Underground, wher...   28K     679     4.9K   
4  A 2D metroidvania with an emphasis on close co...   21K    2.4K     8.3K   

  Wishlist  
0     4.8K  
1     3.6K  
2     2.6K  
3     1.8K  
4     2.3K